src/pretix/control/forms/mailsetup.py

```diff
@@ -54,6 +56,15 @@ class SMTPMailForm(SettingsForm):
     smtp_password = SecretKeySettingsField(
         label=_("Password"),
         required=False,
+        validators=[RegexValidator(
+            r"^[A-Za-z0-9!\"#$%&'()*+,-./:;<=>?@\^_`{}|~]+$",
+            message=format_lazy(
+                _("The password contains characters not supported by our email system. Please only use characters "
+                  "A-Z, a-z, 0-9, and common special characters ({characters})."),
+
+                characters=r'!"#$%%&\'()*+,-./:;<=>?@\^_`{}|~'
```

Comment:
Did check all these special chars are supported? When reading the [issue on cpython](https://github.com/python/cpython/issues/73936) I found [this linked issue](https://github.com/janeczku/calibre-web/issues/2977) that stated backticks cause an error. I also found that hash (#) seem to cause errors in some sendmail implementations (can’t find the link currently, sorry!).

Reply:
`-` should always be last in a character class, otherwise it specifies a range. However this case, the result is accidentally (or trickily) correct, because
```
>>> ord(",")
44
>>> ord("-")
45
>>> ord(".")
46
```

Reply:
The linked issue that talks about backticks is inconsistent, it gives ` (ASCII character 96) as an example but the error message shows ˋ (unicode character U+02CB, "Modifier Letter Grave Accent") to be the problem. I'm not claiming anyone will understand this, but ASCII backtick should be fine.

I did not test all of the characters, I just went by the "can't handle non-ascii characters" error message and think that would be okay for now. We can always reduce the charset further.